subsys/shell/modules/kernel_service.c

```diff
@@ -46,18 +47,31 @@ static int cmd_kernel_cycles(const struct shell *shell,
 	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
 
-	shell_fprintf(shell, SHELL_NORMAL, "cycles: %u hw cycles\n",
+	shell_print(shell, "cycles: %u hw cycles",
 			k_cycle_get_32());
 	return 0;
 }
 
 #if defined(CONFIG_INIT_STACKS) && defined(CONFIG_THREAD_MONITOR) \
 				&& defined(CONFIG_THREAD_STACK_INFO)
+static void thread_state_to_str(const struct k_thread *thread, const struct shell *shell)
```

Comment:
i was more thinking about `const char *thread_state_to_str(u8_t state)`. That could even be moved at some point to some kernel header (e.g. `kernel_structs.h` where flags are defined) since it would be generic and might be reused by logger, printk or shell.

Reply:
Mmm, return string is non-const. Any combination of flags is possible. Thus string must be given as parameter, which makes usage a bit complicated.

Reply:
>  Any combination of flags is possible

I see in fact that in some places there are `thread->base.thread_state |= _THREAD_DEAD` and `thread->base.thread_state |= _THREAD_ABORTING;` assignments but i wonder if that is correct. State should be exclusive.
@andrewboie  @andyross can you help with that? Is this or assignment a bug or a feature?

Reply:
@nordic-krch Seems like i found one place where state is being set as a combination of flags.
https://github.com/zephyrproject-rtos/zephyr/blob/11e91a83c1f6923a0078834404e238e5b7df3693/arch/x86/core/ia32/float.c#L211
defined at 
https://github.com/zephyrproject-rtos/zephyr/blob/11e91a83c1f6923a0078834404e238e5b7df3693/arch/x86/include/ia32/kernel_arch_data.h#L62

What seems to be kinda strange:
```
/* executing context is interrupt handler */
#define _INT_ACTIVE (1 << 7)
/* executing context is exception handler */
#define _EXC_ACTIVE (1 << 6)
```
```
/* Thread is present in the ready queue */
#define _THREAD_QUEUED (BIT(6))
```
ia32 _EXC_ACTIVE overrides the _THREAD_QUEUED flag. Don't you know is it normal?

Reply:
Hm, on other hand Exception or Interrupt execution on x86 is interrupt normal program execution, so we can assume that both flags are used to indicate t hat we're in this context. Seems like only _EXC_ACTIVE == _THREAD_QUEUED  - is suspicious.